render/src/html/mod.rs

```diff
@@ -0,0 +1,237 @@
+//! Defines the [`Html`] struct that is returned when rendering Unimarkup to HTML.
+
+use crate::render::OutputFormat;
+
+pub mod highlight;
+pub mod render;
+
+#[derive(Debug, Default)]
+pub struct HtmlAttribute {
+    pub name: String,
+    pub value: Option<String>,
+}
+
+#[derive(Debug, Default)]
+pub struct HtmlElement {
+    pub name: String,
+    pub attributes: HtmlAttributes,
+    pub content: Option<String>,
+}
+
+#[derive(Debug, Default)]
+pub struct HtmlAttributes(Vec<HtmlAttribute>);
+#[derive(Debug, Default)]
+pub struct HtmlElements(Vec<HtmlElement>);
+
+#[derive(Debug, Default)]
+pub struct HtmlHead {
+    pub elements: HtmlElements,
+    pub syntax_highlighting_used: bool,
+    pub styles: HtmlAttributes, //TODO: replace with CSS struct
+}
+
+impl HtmlHead {
+    fn merge(&mut self, mut other: Self) {
+        self.elements.append(&mut other.elements);
+        self.styles.append(&mut other.styles);
+        self.syntax_highlighting_used |= other.syntax_highlighting_used;
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct HtmlBody {
+    pub elements: HtmlElements,
+}
+
+#[derive(Debug, Default)]
+pub struct Html {
+    pub head: HtmlHead,
+    pub body: HtmlBody,
+    pub lang: String,
+}
+
+impl Html {
+    pub fn with_head(head: HtmlHead) -> Self {
+        Html {
+            head,
+            ..Default::default()
+        }
+    }
+
+    pub fn with_body(element: HtmlElement) -> Self {
```

Comment:
Should this parameter be `HtmlBody`? Or is there a reason `HtmlElement` is used?

Reply:
Since `HtmlBody` only contains `HtmlElements`, it felt tedious to use `HtmlBody` as parameter.
But it is probably better to change this to `Vec<HtmlElement>`.

Reply:
In that case I would suggest implementing `From<HtmlElement>` or `From<Vec<HtmlElement>`, or renaming this function to `with_elements`.
Function name `with_body` is misleading in this case.